File: spec/php/TypeTernaryOpaqueTest.php
<?php
namespace Kaitai\Struct\Tests;

class TypeTernaryOpaqueTest extends TestCase {
    public function testTypeTernaryOpaque() {
        $r = TypeTernaryOpaque::fromFile(self::SRC_DIR_PATH . "/term_strz.bin");

        $this->assertSame('foo', $r->dif->s1);
        $this->assertSame('bar', $r->dif->s2);
        $this->assertSame('|baz@', $r->dif->s3);
    }
}

File: spec/go/type_ternary_opaque_test.go
package spec

import (
	"runtime/debug"
	"os"
	"testing"
	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	. "test_formats"
	"github.com/stretchr/testify/assert"
)

func TestTypeTernaryOpaque(t *testing.T) {
	defer func() {
		if r := recover(); r != nil {
			debug.PrintStack()
			t.Fatal("unexpected panic:", r)
		}
	}()
	f, err := os.Open("../../src/term_strz.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)
	var r TypeTernaryOpaque
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	tmp1, err := r.Dif()
	if err != nil {
		t.Fatal(err)
	}
	assert.EqualValues(t, 102, tmp1.One)
}
